### PR TITLE
#161254668 follow and unfollow another user profile

### PR DESCRIPTION
**Pull request by TheSteelGuy**
#### What does this PR do?
```Adds functionality for users to follow and unfollow other users ```
#### Description of Task to be completed?
```
an authenticated user hitting api/profiles/<username>/follow/
with POST request should be able to follow the user
with DELETE request should be able to unfollow the user
 ```
#### How should this be manually tested?
$ virtualenv env
$ source env/bin/activate
$ cd env
```
- Clone this repo.
``` 
$ git clone https://github.com/andela/ah-orcas.git
```
- Switch to this branch
``` 
$  git checkout  ft-users-follow_each_other_profile-161254668
```
- Create a .env file and run it. Check sample.env file to configure yours.
```
$ source .env
```
- Run the app
```
$ python manage.py runserver
#### Any background context you want to provide?
```Users cannot follow themselves```
#### What are the relevant pivotal tracker stories?

[#161254668](https://www.pivotaltracker.com/story/show/161254668)  

**Screenshot**
screen shot 2018-11-12 at 09 47 53
screen shot 2018-11-12 at 09 51 55
screen shot 2018-11-12 at 09 52 51

### CODE REVIEW

**Comment by Eyansky** [comment]
Kindly add a screenshot of the task.

File: authors/apps/profiles/views.py [hunk 1]
```diff
@@ -86,3 +88,44 @@ def get(self, request, **kwargs):
...
+        except UserProfile.DoesNotExist:
+            return Response(
+                {'Message': 'No profile with this username was found'},
+                status=status.HTTP_404_NOT_FOUND
+            )
+
+        if follower.pk is followee.pk:
+            return Response({
+                'Message': 'You can only follow others, not yourself'
+            }, status=status.HTTP_409_CONFLICT)
```
Comment (Eyansky) [inline authors/apps/profiles/views.py]: I love that you took the effort to implement one cannopt follow themselves. Nice attention to detail.👌🏾

File: requirements.txt [hunk 2]
```diff
@@ -25,32 +32,35 @@ isort==4.3.4
 itypes==1.1.0
 Jinja2==2.10
 lazy-object-proxy==1.3.1
+mando==0.6.4
 MarkupSafe==1.0
 mccabe==0.6.1
 more-itertools==4.3.0
 oauthlib==2.1.0
 openapi-codec==1.3.2
+Pillow==5.3.0
```
Comment (Eyansky) [inline requirements.txt]: What does pillow do? and where did you implement it?

**Comment by Eyansky** [comment]
The title is not in the correct format.

**Comment by Eyansky** [comment]
Remove the commit messages from the description of the pull request.

**Comment by TheSteelGuy** [comment]
@Eyansky done

File: authors/apps/profiles/views.py [hunk 3]
```diff
@@ -86,3 +88,44 @@ def get(self, request, **kwargs):
...
+    serializer_class = ProfileSerializer
+
+    def follow_unfollow(self, username, request, check, status_):
+        follower = self.request.user.userprofile
+        try:
+            followee = UserProfile.objects.get(user__username=username)
+        except UserProfile.DoesNotExist:
+            return Response(
+                {'Message': 'No profile with this username was found'},
+                status=status.HTTP_404_NOT_FOUND
```
Comment (code-sleuth) [inline authors/apps/profiles/views.py]: change this to no content found status code.

**Pull request merged**